client/src/main/java/org/asynchttpclient/netty/ssl/DefaultSslEngineFactory.java

```diff
@@ -26,13 +26,11 @@
 
 public class DefaultSslEngineFactory extends SslEngineFactoryBase {
 
-    private final SslContext sslContext;
+    private volatile SslContext sslContext;
```

Comment:
To be able to make this final we'd need some sort of FactoryFactory that could create a Factory from the Config. I'm not heartbroken over ~2ns volatile read cost per ssl connection though.